### DISCUSSION TITLE
How to get the current user details in Express(nodejs)?

### DISCUSSION THREAD

**Discussion by leelasairam** [supabase/supabase#12412] [Questions]
I am new to supabase. In client side I can get current user details by use below query:
`const { data: { user } } = await supabase.auth.getUser()

 How can I get current user details in Express(nodejs)?

**Comment by GaryAustin1** [comment]
Depends on what you need to do.  Auth-helpers coordinates between browser and server side:  https://supabase.com/docs/guides/auth/auth-helpers

You could also have your own means to pass jwt or get the user session to set the authorization header (or use getUser(jwt) on the server side.   Make sure on a server that you set persistSession;false in your createClient call.

**Reply by dquang-tran** [reply]
Are there examples/best practices on how to do this? 
Does it makes sense rely on supabaseClient.auth.onAuthStateChange to set authorization header for the backend (express) server domain? So that the jwt used for auth is synced with the jwt that the backend would use to retrieve data from database?

The auth-helpers example supports next.js, but this is a scenario where the domain name of the api server is the same, therefore cookies are shared by default.

**Reply by GaryAustin1** [reply]
There is no connection between a browser client and a server client unless you use auth-helpers or you handle getting the access-token/jwt to the server.  onAuthStateChange will not fire there if the browser client signs in.
